### PR TITLE
karmadactl: add liveness probe into kube-controller-manager

### PR DESCRIPTION
**Pull request by carlory**
**What type of PR is this?**

/kind feature

**What this PR does / why we need it**:

Add liveness probe into kube-controller-manager. 

others have already liveness probe.
.
**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
karmadactl: add liveness probe into the kube-controller-manager component
```

### CODE REVIEW

**Comment by jwcesign** [comment]
I think we need to update all the yaml related to enable liveness probe.

File: pkg/karmadactl/cmdinit/kubernetes/deployments.go [hunk 1]
```diff
@@ -233,6 +233,23 @@ func (i *CommandInitOption) makeKarmadaKubeControllerManagerDeployment() *appsv1
...
+					IntVal: 10257,
+				},
+				Scheme: corev1.URISchemeHTTPS,
+			},
+		},
+		InitialDelaySeconds: 10,
+		FailureThreshold:    8,
+		PeriodSeconds:       10,
+		TimeoutSeconds:      15,
+	}
```
Comment (jwcesign) [inline pkg/karmadactl/cmdinit/kubernetes/deployments.go:251]: can we keep it same with `charts/karmada/templates/kube-controller-manager.yaml`:
```yaml
          livenessProbe:
            failureThreshold: 8
            httpGet:
              path: /healthz
              port: 10257
              scheme: HTTPS
            initialDelaySeconds: 10
            periodSeconds: 10
            successThreshold: 1
            timeoutSeconds: 15
```

Comment (carlory) [reply pkg/karmadactl/cmdinit/kubernetes/deployments.go:251]: @jwcesign the default value of successThreshold is 1

```
   successThreshold	<integer>
     Minimum consecutive successes for the probe to be considered successful
     after having failed. Defaults to 1. Must be 1 for liveness and startup.
     Minimum value is 1.
```

**Review by RainbowMango — APPROVED** [review]
/approve

**Pull request merged**